src/getExecutablePath.cpp

```diff
@@ -96,7 +96,7 @@ std::string getExecutablePath()
 
 std::string getExecutablePath()
 {
-    std::array<int, 4> mib = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
+    int mib = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
```

Comment:
This should be an array. The real fix would be in line 102: `if(sysctl(mib.data(), 4, &buf[0], &size, nullptr, 0) != 0)`

Can you change and test that?

Reply:
Ah, right. I forgot to put int mib[4] when making a PR.

Reply:
Is int mib[4] acceptable? We don't need C++ types, plain array of ints works.

Reply:
In the end there is no difference. I just eliminated all plain C arrays from the code and replaced them with the (usually) safer `std::array` (can use `.length()`). I'd like to keep it that way so e.g. grepping for missed C-Arrays works (and returns nothing)

Any reason to prefer the C array besides it is less typing?